src/java.base/share/classes/sun/net/ftp/impl/FtpClient.java

```diff
@@ -1768,24 +1757,20 @@ public long getSize(String path) throws sun.net.ftp.FtpProtocolException, IOExce
     public Date getLastModified(String path) throws sun.net.ftp.FtpProtocolException, IOException {
         issueCommandCheck("MDTM " + path);
         if (lastReplyCode == FtpReplyCode.FILE_STATUS) {
-            String s = getResponseString().substring(4);
-            return parseRfc3659TimeValue(s);
+            String s = getResponseString();
+            return parseRfc3659TimeValue(s.substring(4, s.length() - 1));
         }
         return null;
     }
 
     private static Date parseRfc3659TimeValue(String s) {
-        Date d = null;
-        for (SimpleDateFormat dateFormat : dateFormats) {
-            try {
-                d = dateFormat.parse(s);
-            } catch (ParseException ex) {
-            }
-            if (d != null) {
-                return d;
-            }
+        Date result = null;
+        try {
+            var d = LocalDateTime.parse(s, RFC3659_DATETIME_FORMAT);
+            result = Date.from(d.atZone(ZoneOffset.systemDefault()).toInstant());
```

Comment:
Should this be ZoneOffset.UTC rather than System default? I thought the date returned by the server were supposed to be in GMT.

Reply:
ZoneOffset.UTC should return GMT as was the case with SimpleDateFormat before the changes.

Reply:
right,  thanks for spotting that! I don't know what I was thinking when used `systemDefault` here and when was also got tricked by my own test 🤦